packages/x-date-pickers/src/AdapterDateFnsJalali/index.ts

```diff
@@ -0,0 +1,61 @@
+import BaseAdapterDateFnsJalali from '@date-io/date-fns-jalali';
+import defaultLocale from 'date-fns-jalali/locale/en-US';
+// @ts-ignore
+import longFormatters from 'date-fns-jalali/_lib/format/longFormatters';
+import { MuiFormatTokenMap, MuiPickerFieldAdapter } from '../internals/models';
+
+const formatTokenMap: MuiFormatTokenMap = {
+  y: 'year',
```

Comment:
Should we just import this from the date-fns file ? Same for the others derived adapters whenever le token are the same.

For the methods it would also be nice not to duplicate, just like on date-io, but it would probably be more complicated.
That's a small additional argument in favor of not keeping the migration phase with some methods here and some on date-io forever.